replication-api/src/main/java/org/codice/ditto/replication/api/QueryService.java

```diff
@@ -0,0 +1,15 @@
+package org.codice.ditto.replication.api;
+
+import java.util.stream.Stream;
+import org.codice.ditto.replication.api.data.Query;
+
+/** A service which fetches saved user {@link Query}s. */
+public interface QueryService {
+
+  /**
+   * Provides a stream for retrieving all saved user queries as {@link Query}s.
+   *
+   * @return all the user queries saved on the system
```

Comment:
✏️ This should mention throwing ReplicationException.

Reply:
I do agree with your previous comment @peterhuffer that using `ReplicationException` here is weird since this is a query interface and not replication specific. 

Even if it had been replication specific, I am never a fan of re-using a base and generic exception like `ReplicationException` to wrap everything and anything. Sometimes, it is much better to define subclasses of that one to be a bit more specific.